File: Java/src/main/java/parrot/AfricanParrot.java
package parrot;

public class AfricanParrot extends Parrot {
    protected int numberOfCoconuts;
    protected final double loadFactor = 9.0;

    protected AfricanParrot(int numberOfCoconuts) {
        super();
        this.numberOfCoconuts = numberOfCoconuts;
    }

    @Override
    public double getSpeed() {
        return Math.max(0, getBaseSpeed() - loadFactor * numberOfCoconuts);
    }

    @Override
    public String getCry() {
        return "Sqaark!";
    }
}

File: Kotlin/src/main/kotlin/parrot/AfricanParrot.kt
package parrot

import kotlin.math.max

class AfricanParrot : Parrot{
    constructor(numberOfCoconuts: Int, voltage: Double, isNailed: Boolean) : super(
        type = ParrotTypeEnum.AFRICAN,
        numberOfCoconuts,
        voltage,
        isNailed
    )
    override val speed: Double = max(0.0, baseSpeed - loadFactor * numberOfCoconuts)
    override val cry: String = "Sqaark!"
}